mozjs/src/rust.rs

```diff
@@ -1154,6 +1159,26 @@ impl Drop for EnvironmentChain {
     }
 }
 
+impl<'a> Handle<'a, StackGCVector<JSVal, js::TempAllocPolicy>> {
+    pub fn at(&self, index: u32) -> Handle<'a, JSVal> {
+        unsafe { Handle::from_raw(HandleValueFromStackGCVector((*self).into(), index)) }
+    }
+
+    pub fn len(&self) -> u32 {
+        unsafe { StackGCVectorValueLength((*self).into()) }
+    }
+}
+
+impl<'a> Handle<'a, StackGCVector<*mut JSString, js::TempAllocPolicy>> {
+    pub fn at(&self, index: u32) -> Handle<'a, *mut JSString> {
+        unsafe { Handle::from_raw(HandleStringFromStackGCVector((*self).into(), index)) }
```

Comment:
question: are we indexing without validating index < len() ? what will happen if If a caller passes an out of bounds index? 

if so what about:

```
pub fn len(&self) -> u32 {  }

    pub fn get(&self, index: u32) -> Option<Handle<'a, JSVal>> {
        if index >= self.len() { return None; }
        Some(unsafe { Handle::from_raw(HandleStringFromStackGCVector((*self).into(), index)) })
    }
```